activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -227,8 +228,17 @@ def walk(left, right)
       end
 
       def find_reflection(klass, name)
-        klass._reflect_on_association(name) or
-          raise ConfigurationError, "Association named '#{ name }' was not found on #{ klass.name }; perhaps you misspelled it?"
```

Comment:
So the message would be: `Can't join to association named 'label'; perhaps you misspelled it?` ?

I do see the value in not writing all this code and duplication/restructuring of values. Although, it seems that the original issue might have wanted to facilitate easier developer resolution of the error because the new error message would tell them which method they were using the association incorrectly with (`includes` or `eager_load`). This would allow them to quickly search their codebase for usage of the method + the association (i.e. knowing to quickly grep/search for something like `includes(:wallabies)` vs. just `wallabies` and then screening the results for the offending call). I might be projecting a bit of my own desired resolution behavior onto that though.